### ISSUE TITLE
I can invite people who are already on Wire

### ISSUE THREAD

**Issue by someoneEsle** [wireapp/wire-android#801]
## The problem

When I go to my contacts, I can see a list that includes:
1. people who are on Wire, with their profile pic and name/nickname
2. people who are not on Wire, with a big "Invite" button
3. people who *are* on Wire, who still display the big "Invite" button as if they didn't have the app

People from group 3 are indistinguishable from people from group 2. They are active users, and they have all signed up with their phones (some of them may have added an email to their accounts, but not all of them). Note that not all of people from group 1 have their counterpart in group 3 -- the two are disjoint. I cannot figure out what people in group 3 have in common :/
It may be worth saying that all of the phone numbers on my phone are saved with their respective country code (+1, +44, etc).

## Environment

* Wire version that exhibits the issue: 2.31.324
* Android OS version used to run Wire: stock Android 7.1.2
* Mobile phone model/manufacturer: Nexus 5X

**Comment by forteller** [comment]
Yeah, I've gotten an invite sms from someone once even though I am registered with my phone number
